dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -230,6 +231,45 @@ object TypedEncoder {
       )
   }
 
+  /** java.time Encoders, Spark uses https://github.com/apache/spark/blob/v3.2.0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala for encoding / decoding. */
+  implicit val timeInstant: TypedEncoder[Instant] = new TypedEncoder[Instant] {
+    def nullable: Boolean = false
+
+    def jvmRepr: DataType = ScalaReflection.dataTypeFor[Instant]
+    def catalystRepr: DataType = TimestampType
+
+    def toCatalyst(path: Expression): Expression =
+      StaticInvoke(
+        DateTimeUtils.getClass,
+        TimestampType,
+        "instantToMicros",
```

Comment:
@jgoday that's the main beast, has some limitations 🤷 but we have to use it to be compat with Spark.